docs/how_to_guides/creating_and_editing_expectations/how_to_create_custom_expectations_for_sqlalchemy.rst

```diff
@@ -3,16 +3,266 @@
+    .. code-block:: python
+
+        import sqlalchemy as sa
+        from great_expectations.dataset import (
+            SqlAlchemyDataset,
+            MetaSqlAlchemyDataset,
+        )
+
+    ``SqlAlchemyDataset`` is the parent class used for executing Expectations using sqlalchemy. Most of the core Expectations are built using decorators defined in ``MetaSqlAlchemyDataset``. These decorators streamline the task of extending Great Expectations with custom Expectation logic.
+
+2. **Define a class inheriting from SqlAlchemyDataset**
+
+    .. code-block:: python
+
+        class CustomSqlAlchemyDataset(SqlAlchemyDataset):
+
+            _data_asset_type = "CustomSqlAlchemyDataset"
+
+    Setting the ``_data_asset_type`` is not strictly necessary, but can be helpful for tracking the lineage of instantiated Expectations and :ref:`Validation Results`.
+
+3. **Within your new class, define Expectations using decorators from MetaSqlAlchemyDataset**
+
+    ``column_map_expectations`` are Expectations that are applied to a single column, on a row-by-row basis.
+
+    The ``@MetaSqlAlchemyDataset.column_map_expectation`` decorator wraps a custom function with all the business logic required to turn it into a fully-fledged Expectation. This spares you the hassle of defining required arguments like ``mostly`` and ``result_format``. Your custom function can focus exclusively on the business logic of passing or failing the Expectation.
+
+    In the simplest case, they could be as simple as one-line lambda functions.
+
+    .. code-block:: python
+
+        @MetaSqlAlchemyDataset.column_map_expectation
+        def expect_column_values_to_be_even(self, column):
+            return (sa.column(column) % 2 == 0)
+
+    The business logic for developing SQL queries using ``MetaSqlAlchemyDataset`` deocrators is a little diffeent from ``MetaPandasDataset``. To use the ``column_map_expectation`` decorator, both require that your custom function accept at least two arguments: ``self`` and ``column``. Both create a function API where the user passes a string containing the column name to your Expectation.
+    
+    In the case of ``MetaSqlAlchemyDataset``, the column name string will be passed through to your custom function unaltered. From there, you can use sqlalchemy's ``sqlalchemy.column`` method to create a ``ColumnClause`` based on it. Your function must return a valid ``ColumnClause``, which the decorator will combine with other SqlAlchemy methods to covert your logic into valid SQL queries and wrap the results as an :ref:``ExpectationValiadationResult``.
+
+    By convention, ``column_map_expectations`` always start with ``expect_column_values_...`` or ``expect_column_value_...`` (Ex: ``expect_column_value_word_counts_to_be_between``). Following this pattern is highly recommended, but not strictly required. If you want to confuse yourself with bad names, the package won't stop you.
+
+    |
```

Comment:
?

Reply:
?

I'm not sure what your question is.